azurerm/config.go

```diff
@@ -585,6 +590,15 @@ func (c *ArmClient) registerDatabases(endpoint, subscriptionId string, auth auto
 	c.sqlServersClient = sqlSrvClient
 }
 
+func (c *ArmClient) registerDeviceClients(endpoint, subscriptionId string, auth autorest.Authorizer, sender autorest.Sender) {
+	ihrc := devices.NewIotHubResourceClientWithBaseURI(endpoint, subscriptionId)
+	setUserAgent(&ihrc.Client)
+	ihrc.Authorizer = auth
+	ihrc.Sender = sender
+	ihrc.SkipResourceProviderRegistration = c.skipProviderRegistration
```

Comment:
rather than these 4 lines:

```
setUserAgent(&ihrc.Client)
ihrc.Authorizer = auth
ihrc.Sender = sender
ihrc.SkipResourceProviderRegistration = c.skipProviderRegistration
```

can we swap this over to using the newer style registration function? e.g.

```
c.configureClient(&ihrc.Client, auth)
```